parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -173,7 +173,10 @@ private void initializeInternalReader(ParquetInputSplit split, Configuration con
       }
     }
 
-    if (!reader.getRowGroups().isEmpty()) {
+    if (!reader.getRowGroups().isEmpty() &&
+      // Encrypted files (parquet-mr 1.12+) can't have the delta encoding problem (resolved in parquet-mr 1.8)
```

Comment:
Could we add a test for this?

Reply:
- with delta encoding problem: basically impossible to reproduce :), it was resolved in 1.8
- without this problem: I've had a look at the existing unitests, unfortunately none can be used as a basis for adding a function for this particular situation. This will require building a new unitest from scratch. However, given that a) the patch is small and straightforward b) Spark stopped using this parquet read path - building a full unitest can be an overkill. But if you have a different opinion, please let me know.

Reply:
Thanks for the explanation, I'm fine with leaving out a unit test. Just curious if it would be easy to modify existing tests to make sure that we hit the code.